workers/unity/Packages/io.improbable.gdk.tools/.CodeGenTemplate/CodeGenerationLib/Jobs/CodegenJob.cs

```diff
@@ -14,24 +15,59 @@ public class IgnoreCodegenJobAttribute : Attribute
 
     public abstract class CodegenJob
     {
-        public List<string> InputFiles = new List<string>();
-        public List<string> OutputFiles = new List<string>();
+        public IReadOnlyList<string> InputFiles => inputFiles;
+        public IReadOnlyList<string> OutputFiles => outputFiles;
+
+        private readonly List<string> inputFiles = new List<string>();
+        private readonly List<string> outputFiles = new List<string>();
         public readonly string OutputDirectory;
 
-        protected readonly Dictionary<string, string> Content = new Dictionary<string, string>();
+        protected Logger logger { get; }
+
+        private readonly Dictionary<string, string> Content = new Dictionary<string, string>();
 
         private IFileSystem fileSystem;
         private readonly DetailsStore detailsStore;
 
         public CodegenJob(string baseOutputDirectory, IFileSystem fileSystem, DetailsStore detailsStore)
         {
+            logger = LogManager.GetLogger(GetType().FullName);
+
             OutputDirectory = baseOutputDirectory;
             this.fileSystem = fileSystem;
             this.detailsStore = detailsStore;
         }
 
+        protected void AddInputFiles(IEnumerable<string> inputFilePaths)
+        {
+            foreach (var inputFilePath in inputFilePaths)
+            {
+                AddInputFile(inputFilePath);
+            }
+        }
+
+        protected void AddInputFile(string inputFilePath)
+        {
+            inputFiles.Add(inputFilePath);
+            logger.Trace($"Added input file: {inputFilePath}");
+        }
+
+        protected void AddOutputFile(string outputFilePath)
```

Comment:
If we add an `IEnumerable<string>` method for this (like the input files) we can probably simplify some of the logic in the generators